### PR TITLE
docs: update config docs

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: rel/emqx_conf.template.en.md [hunk 1]
```diff
@@ -7,22 +7,27 @@ and a superset of JSON.
...
 
-1. Cluster configs: `$EMQX_NODE__DATA_DIR/configs/cluster.hocon`
-2. `emqx.conf` + `EMQX_` prefixed environment variables.<br/>
-   Changes in this layer require a full node restart to take effect.
+1. Cluster-synced configs: `$EMQX_NODE__DATA_DIR/configs/cluster.hocon`.
+2. Local node configs: `emqx.conf` + `EMQX_` prefixed environment variables.
 
+:::tip Tip
+Prior to v5.0.23 and e5.0.3, the cluster-synced configs are stored in
+`cluster-override.conf` which on top of the local configs.
```
Comment (id) [inline rel/emqx_conf.template.en.md:15]:
```suggestion
`cluster-override.conf` which is applied on top of the local configs.
```

File: rel/emqx_conf.template.en.md [hunk 2]
```diff
@@ -7,22 +7,27 @@ and a superset of JSON.
...
+
+If upgrade from earlier version, as long as `cluster-override.conf` exists,
```
Comment (thalesmg) [inline rel/emqx_conf.template.en.md:17]:
```suggestion
If upgrading from earlier version, as long as `cluster-override.conf` exists,
```

Comment (id) [inline rel/emqx_conf.template.en.md:17]:
```suggestion
If you upgrade from earlier version, as long as `cluster-override.conf` exists,
```

**Review by thalesmg — DISMISSED** [review]

File: rel/emqx_conf.template.en.md [hunk 3]
```diff
@@ -7,22 +7,27 @@ and a superset of JSON.
...
 The `cluster.hocon` file is overwritten at runtime when changes
-are made from dashboard UI, management HTTP API, or CLI. When clustered,
+are made from Dashboard, management HTTP API, or CLI. When clustered,
 after EMQX restarts, it copies the file from the node which has the greatest `uptime`.
 
 :::tip Tip
-Some of the configs (such as `node.name`) are boot-only configs and not overridable.
-Config values from `cluster.hocon` are **not** mapped to boot configs for
-the config fields attributed with `mapping: path.to.boot.config.key`
+To avoid confusion, do not overlap configs in `cluster.hocon` and `emqx.conf`.
```
Comment (id) [inline rel/emqx_conf.template.en.md:30]:
```suggestion
To avoid confusion, do add the same configs in `cluster.hocon` and `emqx.conf`.
```

Comment (thalesmg) [reply rel/emqx_conf.template.en.md:30]: I think this one should indeed be "don't add the same keys on both", as `cluster.hocon` will be overwritten after most config updates, possibly losing the manual changes.

Comment (id) [reply rel/emqx_conf.template.en.md:30]: Thanks, missed the `not`

File: rel/emqx_conf.template.en.md (hunk 2 above)
Comment (zmstone) [reply rel/emqx_conf.template.en.md]: changed to: If upgraded from an earlier version.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**